applications/luci-app-p910nd/htdocs/luci-static/resources/view/p910nd.js

```diff
@@ -0,0 +1,215 @@
+		s.sortable = true;
+		s.rowcolors = true;
+		s.addbtntitle = _('Add printer config');
+		s.modaltitle = _('Settings');
+
+		o = s.option(form.Flag, 'enabled', _('Enabled'));
+		o.default = true;
+
+		o = s.option(form.Value, 'device', _('Device'),
+			_('Note: character device assignment can change upon reboot/reconnect with multiple USB devices.') + '<br />' +
+			_('Connected %s devices show in this list.').format('<code>lpX</code>'));
+		o.placeholder = '/dev/usb/lp*';
+		if (data[2]) {
+			const devs = data[2].trim().split('\n').sort();
+			for (const chardev of devs) {
+				o.value(chardev);
+			}
+		}
+		o.rmempty = true;
+
+		o = s.option(form.ListValue, 'port', _('Port'),
+			_('Local TCP listen port for this printer.') + '<br />' +
+			_('Be aware: node_exporter also listens on port 9100.'));
+		for (var i = 0; i < 10; i++) {
+			o.value(i, 9100+i);
+		}
+		o.rmempty = true;
+
+		o = s.option(form.Value, 'bind', _('Listen IP'),
+			_('Listen on a specific IP.'));
+		o.datatype = 'ipaddr("nomask")';
+		o.placeholder = _('any');
+		var net_devices = data[1];
+		net_devices.forEach(net_dev => {
+			net_dev.getIPAddrs().forEach(addr => o.value(addr.split('/')[0], E([], [addr.split('/')[0], ' (', E('strong', {}, net_dev.getName()), ')'])));
+			net_dev.getIP6Addrs().forEach(addr => o.value(addr.split('/')[0], E([], [addr.split('/')[0], ' (', E('strong', {}, net_dev.getName()), ')'])));
+		});
+
+		o = s.option(form.Flag, 'bidirectional',
+			_('Bidirectional mode'),
+			_('Whether this print port is bi-directional.'));
+		o.modalonly = true;
+
+		o = s.option(form.Flag, 'runas_root', _('Root'),
+			_('Run the daemon as root; overrides running as %s.').format('<code>user p910nd, group lp</code>'));
+		o.modalonly = true;
+
+		o = s.option(form.Flag, 'mdns', _('mDNS'),
+			_('Whether to advertise this printer via mDNS/Bonjour/ZeroConf.') + '<br />' +
+			_('Note: mDNS only advertises one (the first) printer on this host.'));
+
+		/* See https://developer.apple.com/bonjour/printing-specification/ ( -> bonjourprinting-1.2.1.pdf ) for
+		 * a description of the Bonjour fields a la Apple.
+		 */
+		o = s.option(form.Value, 'mdns_ty', _('Description'),
+			_('User readable description of maker and model.'));
+		o.placeholder = 'HP Color LaserJet CP2025dn';
+		o.rmempty = true;
+		o.optional = true;
+		o.depends({mdns: '1'});
+
+		o = s.option(form.Value, 'mdns_note', _('Location'));
+		o.placeholder = 'By the router';
+		o.rmempty = true;
+		o.optional = true;
+		o.modalonly = true;
+		o.depends({mdns: '1'});
+
+		o = s.option(form.Value, 'mdns_product', _('Product'),
+			_('the value of the "Product" stored in the driver PPD file (enclosed within parentheses). E.g. %s.').format('<code>(MFG MODEL)</code>'));
+		o.placeholder = '(Color LaserJet CP2025dn)';
+		o.rmempty = true;
+		o.optional = true;
+		o.modalonly = true;
+		o.depends({mdns: '1'});
+		o.validate = function(section_id, value) {
+			if(value[0] == '(' && value[value.length-1] == ')')
```

Comment:
Maybe use the o.write function and just silently add the `()`.
FYI: you can use `value.startsWith('(') && value.endsWith(')')`

Reply:
Yeah, thanks, I was thinking about something similar but didn't get around to it.